frontend/src/components/HighlightSearchText.tsx

```diff
@@ -2,74 +2,195 @@
 import { css } from '@emotion/css'
 import { Button } from '@patternfly/react-core'
 import { parseLabel } from '../resources/utils'
-import { MouseEventHandler } from 'react'
-
-const MAX_LABEL_WIDTH = 28
+import { Truncate } from './Truncate'
 
 const buttonDivClass = css({
   '&:hover > button:after': {
-    borderColor: 'rgb(21, 21, 21) !important',
+    borderColor: 'rgb(21, 21, 21)',
   },
   '&:hover > button': {
-    backgroundColor: 'white !important',
+    backgroundColor: 'white',
   },
 })
 
 const buttonClass = css({
-  padding: '4px 0px !important',
-  lineHeight: '10px !important',
+  padding: '4px 0px',
+  lineHeight: '10px',
   margin: '0 2px',
   minWidth: '16px',
+  // Higher specificity to override PatternFly button styles
+  '&.pf-v5-c-button.pf-m-plain': {
+    padding: '4px 0px',
+    lineHeight: '10px',
+  },
 })
 
 const highlightClass = css({
   color: 'var(--pf-v5-global--link--Color)',
   textDecoration: 'underline',
   background: 'none',
   fontWeight: 600,
+  // Higher specificity to ensure underline is preserved
+  '&[data-highlight="true"]': {
+    textDecoration: 'underline',
+  },
+})
+
+const linkClass = css({
+  color: 'var(--pf-v5-global--link--Color)',
+  textDecoration: 'underline',
+  background: 'none',
+  fontWeight: 'normal',
+  // Higher specificity to ensure underline is preserved
+  '&[data-link="true"]': {
+    textDecoration: 'underline',
+  },
 })
 
+// Helper function to create highlighted text segments
+const createHighlightedSegments = (text: string, searchText: string) => {
```

Comment:
hey Randy-- so it looks like you're using this instead of getSlicedText() to get the searched text to highlight but did you need to do that for the truncate/highlight bug
since we're doing a fuzzy search, if you type in something like 'oper' it's going to find strings that have 'ope' in them but now we won't highlight that 'ope' --if you needed to make the change I could live the fine-grained highliting :)

this is a lot simpler

Reply:
thanks Randy--this works great